tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_persistence.c

```diff
@@ -130,9 +370,422 @@ static void test_provisioning_data_load(void)
+					      TEST_VND_MOD_ID, TEST_VND_COMPANY_ID,
+					      &status);
+	if (err || status) {
+		FAIL("Mod sub del all failed (err %d, status %u)", err, status);
+	}
+
+	memcpy(&pub_params, &(struct bt_mesh_cfg_mod_pub)TEST_VND_MOD_PUB_PARAMS,
+	       sizeof(struct bt_mesh_cfg_mod_pub));
+	pub_params.addr = BT_MESH_ADDR_UNASSIGNED;
+	pub_params.uuid = NULL;
+	err = bt_mesh_cfg_mod_pub_set_vnd(test_netkey_idx, TEST_ADDR, TEST_ADDR, TEST_VND_MOD_ID,
+					  TEST_VND_COMPANY_ID, &pub_params, &status);
+	if (err || status) {
+		FAIL("Mod pub set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_model_data_store(test_vnd_model, true, TEST_VND_MOD_DATA_NAME, NULL, 0);
+	if (err) {
+		FAIL("Vnd mod data erase failed (err %d)", err);
+		return;
+	}
+
+	k_sleep(K_SECONDS(CONFIG_BT_MESH_STORE_TIMEOUT));
+
+	PASS();
+}
+
+static void test_cfg_save(void)
+{
+	uint8_t transmit;
+	uint8_t status;
+	int err;
+
+	ASSERT_TRUE(test_preset >= 0 && test_preset <= 1);
+
+	settings_test_backend_clear();
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+
+	if (test_persistence_prov_setup()) {
+		FAIL("Mesh setup failed. Settings should not be loaded.");
+	}
+
+	err = bt_mesh_cfg_beacon_set(test_netkey_idx, TEST_ADDR,
+				     test_cfg_presets[test_preset].beacon, &status);
+	if (err || status != test_cfg_presets[test_preset].beacon) {
+		FAIL("Beacon set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_cfg_ttl_set(test_netkey_idx, TEST_ADDR,
+				  test_cfg_presets[test_preset].ttl, &status);
+	if (err || status != test_cfg_presets[test_preset].ttl) {
+		FAIL("TTL set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_cfg_gatt_proxy_set(test_netkey_idx, TEST_ADDR,
+					 test_cfg_presets[test_preset].gatt_proxy, &status);
+	if (err || status != test_cfg_presets[test_preset].gatt_proxy) {
+		FAIL("GATT Proxy set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_cfg_friend_set(test_netkey_idx, TEST_ADDR,
+				     test_cfg_presets[test_preset].friend, &status);
+	if (err || status != test_cfg_presets[test_preset].friend) {
+		FAIL("Friend set failed (err %d, status %u)", err, status);
+	}
+
+	err = bt_mesh_cfg_net_transmit_set(test_netkey_idx, TEST_ADDR,
+					   test_cfg_presets[test_preset].net_transmit,
+					   &transmit);
+	if (err || transmit != test_cfg_presets[test_preset].net_transmit) {
+		FAIL("Net transmit set failed (err %d, trasmit %x)", err, transmit);
+	}
+
+	err = bt_mesh_cfg_relay_set(test_netkey_idx, TEST_ADDR,
+				    test_cfg_presets[test_preset].relay.state,
+				    test_cfg_presets[test_preset].relay.transmit,
+				    &status, &transmit);
+	if (err || status  != test_cfg_presets[test_preset].relay.state ||
+	    transmit != test_cfg_presets[test_preset].relay.transmit) {
+		FAIL("Relay set failed (err %d, status %u, transmit %x)", err, status,
+		     test_cfg_presets[test_preset].relay.transmit);
+	}
+
+	k_sleep(K_SECONDS(CONFIG_BT_MESH_STORE_TIMEOUT));
```

Comment:
How about checking for "Health Fast Period Divisor" and "Fault" state persistence?

Reply:
That is part of NCSDK-10266.

Reply:
I just checked and it seems like there is no persistence for health model yet.